release_notes.md

```diff
@@ -1,3 +1,82 @@
+- The top-level directory under which these run directories are placed is controlled with the `OPENMDAO_WORKDIR` environment variable, which defaults to the current working directory.
+
+** Note that these changes can cause a huge number of output directories to be placed into the users hard drive when running many OpenMDAO problems, such as when running tests **
+
+We recommend users run tests using temporary directories (see our @use_tempdirs decorator for tests).
+
+Additionally, we now provide an `openmdao.api.clean_outputs()` method as well as a command line form `openmdao clean` or `python -m openmdao clean`.  See the help for these commands for more information, but they'll make your life a bit less chaotic.
+
+In other highlights...
+
+- We now support of numpy 2.0.
+- A new implicit component, `InputResidsComp` has been added. This is probably the simplest possible use case for an implicit component, where the residuals are given as inputs.
+- We've added a new `setup_residuals` method in `ImplicitComponent`, which like `setup_partials`, delays the addition of residuals until the shape of inputs and outputs are known.
+- The OpenMDAO command-line tools are accessible via the more pythonic `python -m openmdao` command. We also added a `view_reports` subcommand to quickly bring up the reports associated with a given case.
+- The `add_input` method now allows the user to specify `require_connection=True`, which will result in an error if the input is left unconnected.
+- Python 3.8 reaches end-of-life next month, and our minimum supported Python version is increased to 3.9.
+
+## New Features
+
+- POEM 097 Implementation, Part 1 [#3287](https://github.com/OpenMDAO/OpenMDAO/pull/3287)
+- Added the 'list_vars' method to the Case  object [#3293](https://github.com/OpenMDAO/OpenMDAO/pull/3293)
+- Added InputResidsComp to the list of components. Added `setup_residuals` method for compatibility with dynamic shaping. [#3295](https://github.com/OpenMDAO/OpenMDAO/pull/3295)
+- POEM 097 Implementation - Part 2 [#3298](https://github.com/OpenMDAO/OpenMDAO/pull/3298)
+- POEM 097 - Part 3 [#3314](https://github.com/OpenMDAO/OpenMDAO/pull/3314)
+- Added `openmdao clean` command line utility and `openmdao.api.clean_outputs` function. [#3318](https://github.com/OpenMDAO/OpenMDAO/pull/3318)
+- Added better error msg when a distributed variable is part of an invalid connection or is unconnected [#3321](https://github.com/OpenMDAO/OpenMDAO/pull/3321)
+- Added code linting with Ruff to the test workflow [#3323](https://github.com/OpenMDAO/OpenMDAO/pull/3323)
+- Added a pre-commit configuration to enable automatic linting with ruff [#3326](https://github.com/OpenMDAO/OpenMDAO/pull/3326) [#3329](https://github.com/OpenMDAO/OpenMDAO/pull/3329) [#3333](https://github.com/OpenMDAO/OpenMDAO/pull/3333)
```

Comment:
I would suggest that these two bullets (ruff & pre-commit) are probably `Miscellaneous` vs `New Features`